File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/actors/buffs/Ooze.java

package com.watabou.pixeldungeon.actors.buffs;

import com.nyrds.pixeldungeon.ml.R;
import com.nyrds.platform.util.StringsManager;
import com.watabou.pixeldungeon.Dungeon;
import com.watabou.pixeldungeon.ResultDescriptions;
import com.watabou.pixeldungeon.ui.BuffIndicator;
import com.watabou.pixeldungeon.utils.GLog;
import com.nyrds.util.Utils;

public class Ooze extends Buff {

	@Override
	public int icon() {
		return BuffIndicator.OOZE;
	}

	@Override
	public boolean act() {
		if (target.isAlive()) {
			target.damage( level, this );
			if (!target.isAlive() && target == Dungeon.hero) {
				Dungeon.fail( Utils.format( ResultDescriptions.getDescription(ResultDescriptions.Reason.OOZE), Dungeon.depth ) );
                GLog.n(StringsManager.getVar(R.string.Ooze_Death), name() );
			}
			spend( TICK );
		}
		if (target.level().water[target.getPos()]) {
			if (level--<= 0) {
				detach();
			}
		}
		return true;
	}
}

File: scripts/buffs/Ooze.lua
---
--- Ooze: buffLevel damage per tick, washed off in water (was actors/buffs/Ooze.java)
---
local RPD  = require "scripts/lib/commonClasses"
local buff = require "scripts/lib/buff"

local CharSprite = luajava.bindClass("com.watabou.pixeldungeon.sprites.CharSprite")

return buff.init{
    desc  = function ()
        return {
            icon          = 8, -- BuffIndicator.OOZE
            name          = "OozeBuff_Name",
            info          = "OozeBuff_Info",
        }
    end,

    act = function(self, buff)
        local target = buff.target

        if target:isAlive() then
            target:damage(buff:level(), buff)
            if not target:isAlive() and target == RPD.Dungeon.hero then
                RPD.Dungeon:fail(RPD.JavaUtils:format(
                    RPD.ResultDescriptions:getDescription(RPD.ResultReason.OOZE), { RPD.Dungeon.depth }))
                RPD.GLog:n(RPD.JavaUtils:format(RPD.textById("Ooze_Death"), { buff:name() }), {})
            end
            buff:spend(1) -- Actor.TICK
        end

        if RPD.Dungeon.level.water[target:getPos() + 1] then
            local lvl = buff:level()
            if lvl <= 0 then
                buff:detach()
            end
            buff:level(lvl - 1)
        end
    end
}
